Android: Use the correct value for notification

srifqi opened the pull request:
**Goal of the PR**
This PR fixes the settings of the notification channel on Android.

**How does the PR work?**
This PR uses the correct field of the notification channel on Android. Its effect is that there will be two notification channels (the old one and the new one) if Minetest is updated from the previous version.

The notification channel creation is also moved into the `MainActivity` class in preparation of other notifications.

**Does it resolve any reported issue?**
The issue has not been reported. See the comparisons below.

**Does this relate to a goal in the roadmap?**
Internal code refactoring and UX improvements

## To do
This PR is Ready for Review.

## How to test
1. Install version 5.8.0 or `master` of Minetest.
2. Open Minetest and close it.
3. Open Minetest's App info.
4. Open Notifications (App notifications) > Notification categories.
6. Open the only category in the list.
7. Check the values of each field.
8. Repeat for this PR version of Minetest.

### Test results
The test was run on Samsung Galaxy A72 (Android 13).

| Menu | Before | After |
|:----:|:------:|:-----:|
| Category List | ![category list before](https://github.com/minetest/minetest/assets/4017302/828193a0-defa-46a8-96d2-15799199ad7b) | ![category list after](https://github.com/minetest/minetest/assets/4017302/d4cf7c5a-33f1-4e6f-9010-5fbc3847c7fe) |
| Category Information | ![category information before](https://github.com/minetest/minetest/assets/4017302/4d613305-0c31-4e59-8f2a-51ad1b4af9d7) | ![category information after](https://github.com/minetest/minetest/assets/4017302/714e8043-6dc2-49de-a238-8640d92161a7) |

sfan5 commented on android/app/src/main/java/net/minetest/minetest/MainActivity.java:
```diff
@@ -114,6 +124,31 @@ private void startNative() {
+		if (notifyManager == null)
+			return;
+
+		String channelId = "net.minetest.minetest";
```
ideally this would be a `public static final` so the magic value exists only once
but doesn't matter too much

srifqi replied:
Done.

sfan5 commented on android/app/src/main/java/net/minetest/minetest/UnzipService.java:
```diff
@@ -99,28 +100,13 @@ protected void onHandleIntent(Intent intent) {
-				mNotifyManager.createNotificationChannel(mChannel);
-			}
-			builder = new Notification.Builder(this, channelId);
+			builder = new Notification.Builder(this, "net.minetest.minetest");
```
should work, no?
```suggestion
			builder = new Notification.Builder(this, MainActivity.NOTIFICATION_CHANNEL_ID);
```

srifqi replied:
Ah, I missed that part. Done.

grorp reviewed (commented):
If I understand correctly, the goal of this PR is to fix the notification channel name and description that are visible in the settings.

Why does this PR create a new notification channel / change the notification channel ID for that? As far as I understand, it should be possible to update the existing notification channel: https://stackoverflow.com/a/48569349

That would also solve the problem with two notification channels appearing in the settings.

grorp commented on android/app/src/main/java/net/minetest/minetest/MainActivity.java:
```diff
@@ -114,6 +126,30 @@ private void startNative() {
+			return;
+
+		String channelName = "General notification";
+		String channelDescription = "Notifications from Minetest";
```
Can these strings be made translatable?

srifqi replied:
Do we want Minetest to react to [ACTION_LOCALE_CHANGED](https://developer.android.com/reference/android/content/Intent#ACTION_LOCALE_CHANGED) intents (most apps) or should the notification channel information be changed when the app started (simpler)?

grorp replied:
> Do we want Minetest to react to [ACTION_LOCALE_CHANGED](https://developer.android.com/reference/android/content/Intent#ACTION_LOCALE_CHANGED) intents (most apps) or should the notification channel information be changed when the app started (simpler)?

Both approaches are fine by me.

srifqi replied:
I decided to just use a simple `getString()`. The ability to react to the intent can be done in another time.

srifqi commented:
> Why does this PR create a new notification channel / change the notification channel ID for that?

The current notification channel ID is "Minetest channel" and the one used in this PR is "net.minetest.net". The reasoning behind it is that the current code swaps between channel ID and channel name. Should we keep use the current channel ID?

| Field | Current | This PR |
| ----- | ------- | ------- |
| Channel ID | Minetest channel | net.minetest.minetest |
| Channel Name | net.minetest.minetest | General notification |
| Channel Description | notifications from Minetest | Notifications from Minetest |

srifqi commented:
The notification channel ID is now the same as before (unchanged). The name and description of the notification channel are now in the strings resource file.

I rebased and squashed this PR, too.

sfan5 commented on android/app/src/main/java/net/minetest/minetest/MainActivity.java:
```diff
@@ -114,6 +126,30 @@ private void startNative() {
+		NotificationChannel notifyChannel;
+		notifyChannel = notifyManager.getNotificationChannel(NOTIFICATION_CHANNEL_ID);
+		if (notifyChannel != null)
+			return;
```
I think you can just safely always create the channel
this is also needed so it works correctly when the language changes.

srifqi replied:
Oh, that is true. Fixed.
